Statistics/Distribution/DiscreteUniform.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE DeriveDataTypeable, DeriveGeneric #-}
+-- |
+-- Module    : Statistics.Distribution.DiscreteUniform
+-- Copyright : (c) 2016 André Szabolcs Szelp
+-- License   : BSD3
+--
+-- Maintainer  : a.sz.szelp@gmail.com
+-- Stability   : experimental
+-- Portability : portable
+--
+-- The discrete uniform distribution. There are two variants of
+-- this distribution. First is the probability distribution on an
+-- inclusive interval {1, ..., n}. This is parametrized with n only,
+-- where p_1, ..., p_n = 1/n. ('discreteUniform').
+--
+-- The second variant is the uniform distribution on {a, ..., b} with
+-- probabilities p_a, ..., p_b = 1/(a-b+1). This is parametrized with
+-- /a/ and /b/. ('discreteUniformAB')
+
+module Statistics.Distribution.DiscreteUniform
+    (
+      DiscreteUniform
+    -- * Constructors
+    , discreteUniform
+    , discreteUniformAB
+    -- * Accessors
+    , rangeFrom
+    , rangeTo
+    ) where
+
+import Data.Aeson (FromJSON, ToJSON)
+import Data.Binary (Binary)
+import Data.Data (Data, Typeable)
+import GHC.Generics (Generic)
+import qualified Statistics.Distribution as D
+import Data.Binary (put, get)
+import Control.Applicative ((<$>), (<*>))
+
+
+-- | The discrete uniform distribution.
+data DiscreteUniform = U {
+      rangeFrom  :: {-# UNPACK #-} !Int
+    -- ^ /a/, the lower bound of the support {a, ..., b} 
+    , rangeTo    :: {-# UNPACK #-} !Int
+    -- ^ /b/, the upper bound of the support {a, ..., b} 
+    } deriving (Eq, Read, Show, Typeable, Data, Generic)
+
+instance FromJSON DiscreteUniform
+instance ToJSON DiscreteUniform
+
+instance Binary DiscreteUniform where
+    put (U a b) = put a >> put b
+    get = U <$> get <*> get
```

Comment:
`get` should test whether Same problem affects FromJSON instance. It should use discreteUniformAB instead of U. Otherwise it's possible to deserialize invalid distribution

It would be nice to add DiscreteGen instance as well